src/Microsoft.ServiceFabric.Services/Communication/Client/CommunicationClientCache.cs

```diff
@@ -251,30 +256,45 @@ public void CleanupCacheEntries(out int totalNumberOfItems, out int numberOfEntr
                 foreach (var entry in this.cache)
                 {
                     ++totalNumberOfItems;
-                    if (!entry.Value.Semaphore.Wait(this.cacheEntryLockWaitTimeForCleanup))
-                    {
-                        // If the wait cannot be satisfied in a short time, then it indicates usage for
-                        // this cache entry, so it is ok to skip this entry in the current cleanup run.
-                        continue;
-                    }
 
                     if (!entry.Value.IsCommunicationClientValid())

```

Comment:
Since the IsCommunicationClientValid changes the CommunicationClientCacheEntry.client, I would assume that the Semaphore needs to be acquired first. I would undo this particular change, leaving the traces and the try-finally.